loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,109 @@
+                for (let i = 0; i < attributeKeys.length; ++i) {
+                    let attributeKey = attributeKeys[i];
+                    let accessor = ArrayItem.Get(`${extensionContext}/attributes/${attributeKey}`, this._loader.gltf.accessors, attributes[attributeKey]);
+                    attributeBufferPromises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attributeKey] = { buffer : data, accessor : accessor };
+                    }));
+                }
+
+                const mesh = ArrayItem.Get(`${context}/mesh`, this._loader.gltf.meshes, extension.mesh || nodeMesh);
+                this._loader._loadMeshAsync(`/meshes/${mesh.index}`, node, mesh, (babylonMeshInstanceNode) => {
+                    // read all the attributes
+                    Promise.all(attributeBufferPromises).then(() => {
+                        attributeKeys = Object.keys(attributeBuffers);
+                        if (attributeKeys.length > 0) {
+                            let instanceCount = attributeBuffers[attributeKeys[0]].accessor.count;
+                            let digitLength = instanceCount.toString().length;
+                            // loop through all the instances and create instances, parent to original transform
+                            for (let i = 0; i < instanceCount; ++i) {
+                                let instance = babylonMeshInstanceNode.clone((babylonMeshInstanceNode.name || babylonMeshInstanceNode.id) + "_" + String(i).padStart(digitLength,'0'), babylonTransformNode, true);
+                                if (instance) {
+                                    instance.position = attributeBuffers["TRANSLATION"] ? Vector3.FromArray(attributeBuffers["TRANSLATION"].buffer, i * 3)
+                                                                                        : Vector3.Zero();
+                                    instance.rotationQuaternion = attributeBuffers["ROTATION"] ? Quaternion.FromEulerAngles(attributeBuffers["ROTATION"].buffer[i * 3], attributeBuffers["ROTATION"].buffer[i * 3 + 1], attributeBuffers["ROTATION"].buffer[i * 3 + 2]) //Quaternion.FromArray(attributeBuffers["ROTATION"].buffer, i * 4) //
+                                                                                                : Quaternion.Identity();
+                                    instance.scaling = attributeBuffers["SCALE"] ? Vector3.FromArray(attributeBuffers["SCALE"].buffer, i * 3)
+                                                                                    : Vector3.One();
+                                    instance.computeWorldMatrix(true);
+                                }
+                            }
+                        }
+                        babylonMeshInstanceNode.setParent(babylonTransformNode);
```

Comment:
Since we are instantiating an additional copy of the source and parent node meshes, I think we'd need some way to properly clean up the scene at export. Currently this implementation adds an additional instance to serve as the instanced copys "source". I'm not sure if there is a cleaner way of handling this case, as we are not guaranteed that the mesh of the node containing this extension is the same as the mesh that will be used for instancing.

Reply:
I decided to reuse the initial source mesh created as a result of `_loadmeshAsync` as the first mesh instance.